crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -1691,4 +1692,31 @@ mod tests {
         stage.run(&mut world);
         assert_eq!(*world.get_resource::<usize>().unwrap(), 1);
     }
+
+    #[test]
+    fn change_ticks_wrapover() {
+        const MAX_TIME_SINCE_LAST_CHECK: u32 = u32::MAX / 8;
+        const MAX_DELTA: u32 = (u32::MAX / 4) * 3;
+
+        let mut world = World::new();
+        world.spawn().insert(0usize);
+        *world.change_tick.get_mut() += MAX_DELTA + 1;
+
+        let mut stage = SystemStage::parallel();
+        fn work() {}
+        stage.add_system(work.system());
+
+        // Overflow twice
+        for _ in 0..10 {
+            stage.run(&mut world);
+            for tracker in world.query::<ChangeTrackers<usize>>().iter(&world) {
```

Comment:
This isn't really testing a user facing api. Its testing an arbitrary "less than or equal to MAX_DELTA" assertion.

I think what we really need to test is that public facing "change detection" apis work when the world tick wraps around (and probably various other states like u32::MAX, u32::MAX/2, 0, 1, etc)

Reply:
(and various states of "last_change_tick" values)

Reply:
Imo this should happen using `Changed<usize>` query filters (and assertions on what they return)

Reply:
Added in 344ae7ba5d537d2c3743f189bccd2ac35fdde000. It only tests a few cases, but it's designed to be trivial to add more (I'm not sure what else should be checked though).
I did not remove the previous test since it also tests `SystemStage::check_change_ticks()`.

Reply:
Awesome. That looks good to me!